src/api/httpClient.ts

```diff
@@ -110,15 +110,24 @@ export class AxiosHttpClient implements HttpClient {
 
         const errorResponse = e.response?.data as ResponseData<T>;
         const error: ErrorResponse | null | undefined = errorResponse.error;
-        if (!error || !error.message || !error.code) {
+        if (!errorResponse && (!error || !error.message || !error.code)) {
           return {
             error: GENERIC_ERROR,
           };
         }
+        if (typeof errorResponse === 'string') {
```

Comment:
When will this happen? errorResponse should always be object and not string

Reply:
This is the response I get from rust api. e.g. calling join context api.
`Err(err) => err.into_response(),`

Reply:
But this should map it to api error type` .map_err(|err| parse_api_error(err));`

Reply:
Test it, it returns string

Reply:
Fixed https://github.com/calimero-network/core/pull/523

Reply:
Now it returns object {error: "message"}  as you can see in the SS. 
<img width="257" alt="Screenshot 2024-07-22 at 15 34 03" src="https://github.com/user-attachments/assets/c932b44f-91b3-4bf8-ad60-1432885888dd">

Now it still needs update in this part of code.

Reply:
Oh, let's fix it with this issue:  https://github.com/calimero-network/core/issues/522

Unblocking now.

Reply:
Will do, FE is ready for this updates.